src/seedu/addressbook/Main.java

```diff
@@ -105,15 +106,23 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
-        try {
+    private CommandResult executeCommand(Command command)
+    {
+        try
+        {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }
+        catch (StorageOperationException e) {
             ui.showToUser(e.getMessage());
-            throw new RuntimeException(e);
+            System.out.println("File is read-only. Change file permissions and retry.");
```

Comment:
Good, but ReadOnly is not the only cause. There can also be file in use exception.

Also, `println` does not work in the actual application, should use logging instead.

Reply:
Ok.